### ISSUE TITLE
not working with $httpProvider.interceptors.push

### ISSUE THREAD

**Issue by devmark** [nervgh/angular-file-upload#296]
var csrf_token = document.querySelector('meta[name="csrf-token"]').getAttribute('content');

```
var uploader = $scope.uploader = new FileUploader({
    // your code here...
    headers : {
        'X-CSRF-TOKEN': csrf_token // X-CSRF-TOKEN is used for Ruby on Rails Tokens
    },
    //...
});
```

bcz i have to push auth code with upload image.
$httpProvider.interceptors.push('interceptor'); not work with this package

**Comment by kikar** [comment]
Same for me, I can't attach a JWT if interceptor don't work. Can you fix this?

**Comment by Mr-Anonymous** [comment]
I am having a similar issue. I have set-up JWT using Satellizer which adds the JWT Token in headers, But when I use `uploadAll()`, it  doesnt work and I am receiving `invalid_token` error and I believe its because JWT token is not passed with `uploader`. Have you found a solution for this please?

**Comment by Mr-Anonymous** [comment]
Sorry, didint mean to hijack this issue. I wasnt sure if this is the same as what I am having trouble with. So I created a separate thread for it: https://github.com/nervgh/angular-file-upload/issues/486

**Comment by kalleniklasson** [comment]
A good thing would be to actually let the existing interceptor handle the requests so that you dont need to repeat alot of logic. Since you might have logic in the interceptor to handle refreshtokens and such on authorization rejections

**Comment by ataraxus** [comment]
Hey, is there any workaround for this?
I use interceptors and now angular-file-upload causes this: 
Uncaught Error: [$injector:cdep] Circular dependency found: $http <- $templateFactory <- $view <- $state <- $http <- FileUploader

I'm using version 2.2.0

**Comment by ataraxus** [comment]
Just tried master and it failed in the same way, do I miss something?

**Comment by nervgh** [comment]
Closed in favor of #434
